staging/src/k8s.io/apiextensions-apiserver/test/integration/convert/webhook.go

```diff
@@ -25,91 +25,51 @@ import (
-// case an error is returned.
-type WaitReadyFunc func(timeout time.Duration, triggerConversionFn func() error) error
-
-// StartConversionWebhookServerWithWaitReady starts an http server with the provided handler and returns the WebhookClientConfig
-// needed to configure a CRD to use this conversion webhook as its converter.
-// It also returns a WaitReadyFunc to be called after the CRD is configured to wait until the conversion webhook handler
-// accepts at least one conversion request. If the server fails to start, an error is returned.
-// WaitReady is useful when changing the conversion webhook config of an existing CRD because the update does not take effect immediately.
-func StartConversionWebhookServerWithWaitReady(handler http.Handler) (func(), *apiextensionsv1beta1.WebhookClientConfig, WaitReadyFunc, error) {
-	var once sync.Once
-	handlerReadyC := make(chan struct{})
-	readyNotifyHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		once.Do(func() {
-			close(handlerReadyC)
-		})
-		handler.ServeHTTP(w, r)
-	})
-
-	tearDown, webhookConfig, err := StartConversionWebhookServer(readyNotifyHandler)
-	if err != nil {
-		return nil, nil, nil, fmt.Errorf("error starting webhook server: %v", err)
-	}
-
-	waitReady := func(timeout time.Duration, triggerConversionFn func() error) error {
-		var err error
-		for {
-			select {
-			case <-handlerReadyC:
-				return nil
-			case <-time.After(timeout):
-				return fmt.Errorf("Timed out waiting for CRD webhook converter update, last trigger conversion error: %v", err)
-			case <-time.After(100 * time.Millisecond):
-				err = triggerConversionFn()
-			}
-		}
-	}
-	return tearDown, webhookConfig, waitReady, err
-}
-
 // StartConversionWebhookServer starts an http server with the provided handler and returns the WebhookClientConfig
 // needed to configure a CRD to use this conversion webhook as its converter.
 func StartConversionWebhookServer(handler http.Handler) (func(), *apiextensionsv1beta1.WebhookClientConfig, error) {
-	// Use a unique path for each webhook server. This ensures that all conversion requests
-	// received by the handler are intended for it; if a WebhookClientConfig other than this one
-	// is applied in the api server, conversion requests will not reach the handler (if they
-	// reach the server they will be returned at 404). This helps prevent tests that require a
-	// specific conversion webhook from accidentally using a different one, which could otherwise
-	// cause a test to flake or pass when it should fail. Since updating the conversion client
-	// config of a custom resource definition does not take effect immediately, this is needed
-	// by the WaitReady returned StartConversionWebhookServerWithWaitReady to detect when a
-	// conversion client config change in the api server has taken effect.
-	path := fmt.Sprintf("/conversionwebhook-%s", uuid.NewUUID())
```

Comment:
@jpbetz I believe this was never needed because the test server chooses different ports on each run and therefore reuse of the same URL is highly unlikely or even impossible. Do you agree?

Reply:
agree